### PR TITLE
Exclude context Cancelled

### PR DESCRIPTION
**Pull request by bbernays**
Closes: #364 

Output:

```
cloudquery % go run main.go fetch
Initializing CloudQuery Providers...

✓ cq-provider-aws@v0.8.4 verified    0s  100 %

Finished provider initialization...

Upgrading CloudQuery providers [aws]

✓ Upgraded provider aws to v0.8.4 successfully.

Finished upgrading providers...

Starting provider fetch...

⚠️ cq-provider-aws@latest diagnostics: 18    3s [===========>--------------------------------------------------|  Finished Resources: 23/120
^C2021/12/27 12:26:22 28 Ctrl-c
⚠️ cq-provider-aws@latest diagnostics: 47    3s [============================>---------------------------------|  Finished Resources: 57/120

Fetch Diagnostics for provider aws:

Resource: emr.clusters Type: Access Severity: Warning
        Summary: EMR: ListClusters - User: arn:aws:sts::7049xxxxxxxx:assumed-role/NoPermissions/aws-go-sdk-1640625979772028000 is not authorized to perform: elasticmapreduce:ListClusters on resource: * because no identity-based policy allows the elasticmapreduce:ListClusters action
        Remediation: You are not authorized to perform this operation. Check your IAM policies, and ensure that you are using the correct access keys.
Resource: sns.subscriptions Type: Access Severity: Warning
        Summary: SNS: ListSubscriptions - User: arn:aws:sts::7049xxxxxxxx:assumed-role/NoPermissions/aws-go-sdk-1640625979772028000 is not authorized to perform: SNS:ListSubscriptions on resource: arn:aws:sns:us-east-1:7049xxxxxxxx:* because no identity-based policy allows the SNS:ListSubscriptions action
        Remediation: You are not authorized to perform this operation. Check your IAM policies, and ensure that you are using the correct access keys.

.
.
.

Provider fetch complete.

Provider aws fetch summary:  ✓ Total Resources fetched: 64       ⚠️ Warnings: 48  ❌ Errors: 0
```

### CODE REVIEW

File: pkg/client/client.go [hunk 1]
```diff
@@ -460,7 +461,10 @@ func (c *Client) Fetch(ctx context.Context, request FetchRequest) (res *FetchRes
 			)
 			for {
 				resp, err := stream.Recv()
-				if err == io.EOF {
+				st, ok := status.FromError(err)
+
+				if (ok && st.Code() == gcodes.Canceled) || err == io.EOF {
```
Comment (disq) [inline pkg/client/client.go:466]: if `err` is nil, `ok` will be `true` and `st` will be `nil`, so panic... needs a better `if`

**Review by disq — CHANGES_REQUESTED** [review]

File: pkg/client/client.go (hunk 1 above)
Comment (bbernays) [reply pkg/client/client.go]: Fixed

**Review by disq — APPROVED** [review]

File: pkg/ui/console/client.go [hunk 2]
```diff
@@ -121,7 +123,12 @@ func (c Client) Fetch(ctx context.Context, failOnError bool) error {
 	}
 	response, err := c.c.Fetch(ctx, request)
 	if err != nil {
-		return err
+		// Ignore context cancelled error
+		st, ok := status.FromError(err)
+		if !ok || st.Code() != gcodes.Canceled {
+			return err
+		}
```
Comment (roneli) [inline pkg/ui/console/client.go:130]:
```suggestion
		
		if st, ok := status.FromError(err); !ok || st.Code() != gcodes.Canceled {
			return err
		}
```

File: pkg/client/client.go [hunk 3]
```diff
@@ -460,22 +461,25 @@ func (c *Client) Fetch(ctx context.Context, request FetchRequest) (res *FetchRes
...
-						PartialFetchErrors:    partialFetchResults,
-						FetchErrors:           fetchErrors,
-						FetchResources:        fetchedResources,
-					}
-					return nil
-				}
+
 				if err != nil {
+					st, ok := status.FromError(err)
+					if (ok && st.Code() == gcodes.Canceled) || err == io.EOF {
```
Comment (roneli) [inline pkg/client/client.go:467]: I think this should be in a switch case and messages should be printed differently (i.e provider fetch cancelled/finished/failed), would also add to the provider fetch summary its finish state i.e finish/error/cancelled.

**Review by roneli — CHANGES_REQUESTED** [review]
Had  some suggestions comments :)

File: pkg/client/client.go (hunk 3 above)
Comment (bbernays) [reply pkg/client/client.go:467]: Couldn't figure out how to use a switch without duplicating a bunch of code.

Comment (bbernays) [reply pkg/client/client.go:467]: Output now looks like this when we cancel the invocation
```
Provider aws fetch summary: ❌ (canceled) Total Resources fetched: 24    ⚠️ Warnings: 0   ❌ Errors: 0
```

**Review by roneli — APPROVED** [review]

**Pull request merged**